lighthouse-core/test/report/v2/report-generator-test.js

```diff
@@ -129,9 +129,9 @@ describe('ReportGeneratorV2', () => {
     });
 
     it('should inject the report JSON', () => {
-      const code = 'hax</script><script>console.log("pwned");%%LIGHTHOUSE_JAVASCRIPT%%';
+      const code = 'hax\u2028</script><script>console.log("pwned");%%LIGHTHOUSE_JAVASCRIPT%%';
       const result = new ReportGeneratorV2().generateReportHtml({code});
-      assert.ok(result.includes('"code":"hax'), 'injects the json');
+      assert.ok(result.includes('"code":"hax\\n'), 'injects the json');
       assert.ok(result.includes('\\u003c/script'), 'escapes HTML tags');
```

Comment:
I noticed this line was passing regardless of having a `<` escape on the JSON because we also escape `<` elsewhere in the parts of the full HTML. e.g. `REPORT_JAVASCRIPT` is escaped too, so `Dummy text for ensuring report robustness: </script>` ends up passing this assertion even if the JSON isn't escaped.

Switching it to verifying more of the string at once could work around this

Reply:
ah I gotcha, yeah I added distinct assertions for the json and the javascript 👍